python/lsst/ts/MTAOS/MtaosCsc.py

```diff
@@ -261,8 +374,18 @@ def _getTimestamp(self):
             Data to put in the DDS (data distribution service) topic.
         """
 
-        calibsDir = data.directoryPath
-        await self._runTaskInNewEventLoop(self.model.procCalibProducts, calibsDir)
+        self._logExecFunc()
+        self._checkEnabledState()
+
+        try:
+            calibsDir = data.directoryPath
+            await self._runTaskInNewEventLoop(self.model.procCalibProducts,
+                                              calibsDir)
```

Comment:
I added this is because some calculations in AOS is super long (> 10 sec, some maybe 90 sec although the target is 15 sec). I am not sure other CSC will have the same condition or not. If other CSC has no such condition, maybe we do not need to add this to `BaseCsc`. Keep the code simple (as less public functions as possible) is easier to maintain in the future.